eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/GA4GHVariantCallSetWSServer.java

```diff
@@ -72,24 +80,36 @@ public GASearchCallSetsResponse getCallSets(@RequestParam("variantSetIds") Strin
         if (files.isEmpty()) {
             throw new IllegalArgumentException("The 'variantSetIds' argument must not be empty");
         }
-        
-        VariantSourceDBAdaptor dbAdaptor = DBAdaptorConnector.getVariantSourceDBAdaptor("hsapiens_grch37");
-        
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName("hsapiens_grch37"));
+
         int idxCurrentPage = 0;
         if (pageToken != null && !pageToken.isEmpty() && StringUtils.isNumeric(pageToken)) {
             idxCurrentPage = Integer.parseInt(pageToken);
             queryOptions.put("skip", idxCurrentPage * limit);
         }
         queryOptions.put("limit", limit);
-        
-        List<String> filesList = Arrays.asList(files.split(","));
-        QueryResult<List<String>> qr = dbAdaptor.getSamplesBySources(filesList, queryOptions);
-        
+
+        PageRequest pageRequest = Utils.getPageRequest(queryOptions);
+
+        List<VariantSourceEntity> variantSourceEntities = repository.findByFileIdIn(files, pageRequest);
+        long numTotalResults = repository.countByFileIdIn(files);
+
+        List<String> fileIds = variantSourceEntities.stream()
+                                                    .map(VariantSourceEntity::getFileId)
+                                                    .collect(Collectors.toList());
+
+        List<List<String>> samplesLists  = variantSourceEntities.stream()
+                                                                .map(VariantSourceEntity::getSamplesPosition)
+                                                                .map(Map::keySet)
+                                                                .map(ArrayList::new)
+                                                                .collect(Collectors.toList());
+
         // Convert sample names objects to GACallSet
-        List<GACallSet> gaCallSets = GACallSetFactory.create(filesList, qr.getResult());
+        List<GACallSet> gaCallSets = GACallSetFactory.create(fileIds, samplesLists);
         // Calculate the next page token
         int idxLastElement = idxCurrentPage * limit + limit;
-        String nextPageToken = (idxLastElement < qr.getNumTotalResults()) ? String.valueOf(idxCurrentPage + 1) : null;
+        String nextPageToken = (idxLastElement < numTotalResults) ? String.valueOf(idxCurrentPage + 1) : null;
```

Comment:
I don't know how much the usage of `Pageable` (mentioned above) could also affect this code block, but anyway, could the `nextPageToken` initialization be encapsulated in a utility method as in [the modified EvaWSServer class](https://github.com/jmmut/eva/blob/60ffcf2a22e8ec178508fad5ad4ce9e5a680ec8b/eva-server/src/main/java/uk/ac/ebi/eva/server/ws/EvaWSServer.java)?

Note that change will be merged in the tests PR, it is currently in the `Utils` class.